locked-asset/energy-factory/src/token_merging.rs

```diff
@@ -102,16 +102,29 @@ pub trait TokenMergingModule:
         self.require_not_paused();
 
         let payments = self.get_non_empty_payments();
-        let output_amount_attributes = self.merge_tokens(payments, opt_original_caller);
+        let caller = self.blockchain().get_caller();
+        let original_caller = match opt_original_caller {
```

Comment:
we have a common function for this, no ?

Reply:
Fixed.